numpy/lib/tests/test_function_base.py

```diff
@@ -1498,6 +1500,56 @@ def test_size_zero_output(self):
             f(x)
 
 
+class TestLeaks(object):
+    class A(object):
+        iters = 20
+
+        def bound(self, *args):
+            return np.int(0)
+
+        @staticmethod
+        def unbound(*args):
+            return np.int(0)
+
+        def npy_bound(self, a):
+            return types.MethodType(np.frompyfunc(self.bound, 1, 1), a)
```

Comment:
I'm very confused by what exactly we're trying to test here.

Reply:
`self.bound` is part of a reference cycle that requires a gc collection cycle to break. I added more clarification in the test docstring. Improvements welcome

Reply:
The PR looks good to me, if nobody complains will merge in a bit probably. The test couldbe a bit more obvious to read, or maybe add a trivial one:
```python
def func(a):
    return 0
func.vectorized = np.frompyfunc(func)
```
giving a minimal cycle. The other part would probably be simpler by making it a realistic use case:
```python
class A(object):
    def __init__(self):
        # Wrap bound_method into a ufunc with `np.frompyfunc`. This creates
        # a reference cycle:
        #     self.method -> self   # since it is a bound method (knows "self")
        #     self -> self.wrapped_method  # Set here
        #     self.wrapped_method -> self.method  # Stores self.method to call it
        self.wrapped_method = np.frompyfunc(self.method)

    def method(self):
        return 0
```

Reply:
I'm not following. Where would I add that line?

Reply:
Sorry, I think what confuses me is that I believe you do not need this at all. You can simply remove the `npy_bound` and `npy_unbound` logic and just make it `a.f = np.frompyfunc(func, 1, 1)`.

Reply:
Adopted